server/routes/cart/addToCart.js

```diff
@@ -2,74 +2,64 @@ const express = require("express");
 const router = express.Router();
 const knex = require("../../utils/dbConfig");
 const passport = require("passport");
+const {
+  insertOrderCart,
 
+  getOrderCartBy,
+  insertOrderItems,
+} = require("../../queries/order");
 router.get(
   "/:variant_id",
   passport.authenticate("jwt", { session: false }),
   (req, res) => {
     user_id = req.user.id;
-    if (req.body.quantity === undefined) {
-      req.body.quantity = 1;
+    let quantity = req.body.quantity;
+    if (req.body.quantity === undefined || req.body.quantity === null) {
+      quantity = 1;
     } else {
-      req.body.quantity = parseInt(req.body.quantity);
+      quantity = parseInt(req.body.quantity);
     }
     console.log("req.user", user_id);
-    knex("cart")
-      .where({ "cart.user_id": user_id })
-      .then((row) => {
-        if (row.length === 0) {
-          knex("cart")
-            .insert({ user_id: user_id })
-            .returning("id")
+    console.log("quantity", quantity);
+
+    getOrderCartBy("user_id", user_id).then((row) => {
+      if (row.length === 0) {
+        console.log("row", row);
+        insertOrderCart(user_id, "cart").then((row) => {
+          insertOrderItems(row[0].id, req.params.variant_id, quantity)
             .then((row) => {
+              res.json(row);
+            })
+            .catch((err) => {
+              console.log("err", err);
+              res.status(500).json(err);
+            });
+        });
+      } else {
+        console.log("row1", row);
+        const order_id = row[0].id;
+        knex("order_items")
```

Comment:
this one can also be moved to queries right? 
it will be inconsistent if we move some of the queries and not move the rest. Create a function that will take args to use for where and return response, devs can sent one arg or multiple
what do you think?